geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/AccumuloFeatureIndex.scala

```diff
@@ -10,58 +10,60 @@ package org.locationtech.geomesa.accumulo.index
 import com.typesafe.scalalogging.LazyLogging
 import org.apache.accumulo.core.client.mock.MockConnector
-import org.apache.accumulo.core.data.{Key, Value}
+import org.apache.accumulo.core.conf.Property
+import org.apache.accumulo.core.data.{Key, Range, Value}
 import org.apache.hadoop.io.Text
 import org.geotools.filter.identity.FeatureIdImpl
 import org.locationtech.geomesa.accumulo.data._
-import org.locationtech.geomesa.accumulo.index.attribute.AttributeIndexV2
-import org.locationtech.geomesa.accumulo.index.id.RecordIndexV1
-import org.locationtech.geomesa.accumulo.{AccumuloFeatureIndexType, AccumuloIndexManagerType}
-import org.locationtech.geomesa.utils.index.IndexMode.IndexMode
-
-import scala.util.Try
-import scala.util.control.NonFatal
-// noinspection ScalaDeprecation
-import org.locationtech.geomesa.accumulo.index.attribute.AttributeIndex
-import org.locationtech.geomesa.accumulo.index.z2.{XZ2Index, Z2IndexV1}
-import org.locationtech.geomesa.accumulo.index.z3.{XZ3Index, Z3IndexV1, Z3IndexV2}
-// noinspection ScalaDeprecation
-import org.locationtech.geomesa.accumulo.index.id.RecordIndex
-import org.locationtech.geomesa.accumulo.index.z2.Z2Index
-import org.locationtech.geomesa.accumulo.index.z3.Z3Index
+import org.locationtech.geomesa.accumulo.index.legacy.attribute.{AttributeIndexV2, AttributeIndexV3}
+import org.locationtech.geomesa.accumulo.index.legacy.id.RecordIndexV1
+import org.locationtech.geomesa.accumulo.index.legacy.z2.{Z2IndexV1, Z2IndexV2}
+import org.locationtech.geomesa.accumulo.index.z3.{Z3Index, Z3IndexV1, Z3IndexV2}
 import org.locationtech.geomesa.accumulo.util.GeoMesaBatchWriterConfig
+import org.locationtech.geomesa.accumulo.{AccumuloFeatureIndexType, AccumuloIndexManagerType, AccumuloVersion}
 import org.locationtech.geomesa.features.SerializationOption.SerializationOptions
 import org.locationtech.geomesa.features.{SerializationType, SimpleFeatureDeserializers}
 import org.locationtech.geomesa.security.SecurityUtils
+import org.locationtech.geomesa.utils.index.IndexMode.IndexMode
 import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
 
-// noinspection ScalaDeprecation
+import scala.util.Try
+import scala.util.control.NonFatal
+
 object AccumuloFeatureIndex extends AccumuloIndexManagerType with LazyLogging {
 
-  private val SpatialIndices        = Seq(Z2Index, XZ2Index, Z2IndexV1)
+  val FullColumnFamily      = new Text("F")
+  val IndexColumnFamily     = new Text("I")
+  val BinColumnFamily       = new Text("B")
+  val AttributeColumnFamily = new Text("A")
+
+  val EmptyColumnQualifier  = new Text()
+
+  private val SpatialIndices        = Seq(Z2Index, XZ2Index, Z2IndexV2, Z2IndexV1)
   private val SpatioTemporalIndices = Seq(Z3Index, XZ3Index, Z3IndexV2, Z3IndexV1)
-  private val AttributeIndices      = Seq(AttributeIndex, AttributeIndexV2)
+  private val AttributeIndices      = Seq(AttributeIndex, AttributeIndexV3, AttributeIndexV2)
```

Comment:
new schemas will get the latest attribute index, older schemas will continue to use the older ones. note that it will use whatever SFC is compatible with the schema - Z3, XZ3, Z2 or XZ2.